src/app/examples/stock-market/stock-market.service.ts

```diff
@@ -6,30 +6,26 @@ import { map } from 'rxjs/operators';
 import { Stock } from './stock-market.reducer';
 
 const PROXY_URL = 'https://cors-anywhere.herokuapp.com/';
-const API_URL = 'https://finance.google.com/finance?output=json&q=NYSE:';
 
 @Injectable()
 export class StockMarketService {
   constructor(private httpClient: HttpClient) {}
 
   retrieveStock(symbol: string): Observable<Stock> {
     return this.httpClient
-      .get(PROXY_URL + API_URL + symbol, { responseType: 'text' })
+      .get(PROXY_URL + `https://api.iextrading.com/1.0/stock/${symbol}/stats`)
       .pipe(
-        map((res: string) => JSON.parse(res.replace('//', ''))[0]),
         map((stock: any) => ({
-          symbol: stock.t,
-          exchange: stock.e,
-          last: stock.l,
+          symbol: stock.companyName,
+          exchange: stock.EBITDA,
+          last: stock.latestEPS,
```

Comment:
Unfortunately this is not correct. I tried it with `FB` which is currently around 163 and the `latestEPS` property value is `5.93` also calculating change should be done from the last price not a 30 day average.

As I looked into API response, it unfortunately doesn't provide the numbers we need.

The API can be delayed or not precise, the exact numbers are not important nut it would be very nice if they were at least roughly realistic.